File: sample-rx/src/main/java/com/mateuszkoslacz/moviper/rxsample/viper/routing/UserDetailsRouting.java
package com.mateuszkoslacz.moviper.rxsample.viper.routing;

import android.app.Activity;

import com.mateuszkoslacz.moviper.base.routing.BaseViewHelperRxRouting;
import com.mateuszkoslacz.moviper.rxsample.viper.contract.UserDetailsContract;
import com.mateuszkoslacz.moviper.rxsample.viper.view.activity.FullscreenPhotoActivity;

public class UserDetailsRouting
        extends BaseViewHelperRxRouting<Activity, UserDetailsContract.ViewHelper>
        implements UserDetailsContract.Routing {

    @Override
    public void startFullscreenPhotoActivity(String photoUrl) {
        if (isViewHelperAttached()) {
            FullscreenPhotoActivity.start(getRelatedContext(), photoUrl,
                    getViewHelper().getAvatarImageView());
        }
    }
}

File: sample-super-rx-ai-kotlin/src/main/java/com/mateuszkoslacz/moviper/rxsample/viper/routing/UserDetailsRouting.kt
package com.mateuszkoslacz.moviper.rxsample.viper.routing

import android.app.Activity
import com.mateuszkoslacz.moviper.base.routing.BaseViewHelperRxRouting
import com.mateuszkoslacz.moviper.rxsample.viper.contract.UserDetailsContract
import com.mateuszkoslacz.moviper.rxsample.viper.view.activity.FullscreenPhotoActivity

class UserDetailsRouting : BaseViewHelperRxRouting<Activity, UserDetailsContract.ViewHelper>(), UserDetailsContract.Routing {

    override fun startFullscreenPhotoActivity(photoUrl: String) {
        if (isViewHelperAttached) {
            FullscreenPhotoActivity.start(relatedContext!!, photoUrl,
                    viewHelper!!.avatarImageView)
        }
    }
}